test/storage_test.go

```diff
@@ -327,10 +328,48 @@ var _ = Describe(
 								Expect(err).Should(HaveOccurred())
 							},
 						)
-
 					},
 				)
+				Context(
+					"When configuring workdir", func() {
+						It(
+							"Should allow emptydir volume for default workdir", func() {
+								aeroCluster := createDummyAerospikeClusterWithNonPVWorkdir(
+									clusterNamespacedName, "/opt/aerospike", "",
+								)
+
+								err := deployCluster(
+									k8sClient, ctx, aeroCluster,
+								)
+								Expect(err).ShouldNot(HaveOccurred())
+							},
+						)
+						It(
+							"Should allow emptydir volume for any workdir", func() {
+								aeroCluster := createDummyAerospikeClusterWithNonPVWorkdir(
+									clusterNamespacedName, "/opt/aerospike/data", "/opt/aerospike/data",
+								)
+
+								err := deployCluster(
+									k8sClient, ctx, aeroCluster,
+								)
+								Expect(err).ShouldNot(HaveOccurred())
+							},
+						)
+						It(
+							"Should allow default workdir without any configured volume", func() {
```

Comment:
The server will run but it will not be able to create smd files. I think we can suggest that not-root users use empty dir. In general, default dirs should not be used in production.